### PR TITLE
Methods: Computing DWPCs with matrix multiplication

### PR DESCRIPTION
**Pull request by dhimmel**
Computing DWPCs with matrix multiplication scaffold
Empirical DWPC p-values

### CODE REVIEW

File: content/05.main-text.md [hunk 1]
```diff
@@ -255,6 +255,50 @@ We converted Hetionet v1.0 to HetMat format and uploaded the `hetionet-v1.0.hetm
...
+- BAAB
+- BABA
+- repeat_around
+- interior_complete_group
+- disjoint_groups
+- other
+
+Include information on what lengths of metapath we have completely solved this for. Mention testing against existing methods.
+
+Mention approximations. Also note independent approximation work at https://github.com/mmayers12/hetnet_ml
```
Comment (dhimmel) [inline content/05.main-text.md:286]: @zietzm and @kkloste finally writing up your suite of DWPC methods.

@zietzm IIRC you may have created some long form documentation on the various DWPC methods based on metapath category. Do you remember if you've created any descriptions of the methodology here that we could use for the manuscript?

@zietzm what do you think is the best way for us to describe the suite of DWPC matrix multiplication methods? Formulas or just examples that link to the code implementation?

Comment (zietzm) [reply content/05.main-text.md:286]: > you may have created some long form documentation

I recall doing this, but unfortunately I can't seem to find it. I'm going to keep looking

> what do you think is the best way for us to describe the suite of DWPC matrix multiplication methods? Formulas or just examples that link to the code implementation?

In the main manuscript, my vote is for just a high-level description. But perhaps a supplemental section with formulae and brief descriptions?

Comment (dhimmel) [reply content/05.main-text.md:286]: > I recall doing this, but unfortunately I can't seem to find it. I'm going to keep looking

Some references I'm digging up:

- https://zietzm.com/Vagelos2017/ / https://doi.org/10.6084/m9.figshare.5346577.v1
- https://github.com/greenelab/hetmech/blob/b54ef23b293b4fbf8dd60a06e1994d2c677bde68/data/rephetio-DWPCs-hetmech-runtime.tsv

> In the main manuscript, my vote is for just a high-level description. But perhaps a supplemental section with formulae and brief descriptions?

Let's see what existing resources we dig up. I think these algorithms are a pretty major scientific contribution, so I don't want to hide them in a supplement. On the other hand, fully describing it in the manuscript might get out of hand. Let's think of a succinct but explanatory way to get the different metapath categories and algorithms across.

File: content/05.main-text.md [hunk 2]
```diff
@@ -300,21 +354,49 @@ TODO: somewhere link to notebook https://github.com/greenelab/hetmech/blob/04206
...
 ### Gamma-hurdle distribution
 
+<!-- references
+  https://nbviewer.jupyter.org/github/zietzm/hetmech/blob/afde1c6c2a7dea7db370ab543e0abcb992d34b0b/explore/gamma-hurdle-heatmaps/gamma-heatmaps.ipynb
+  https://github.com/greenelab/hetmech/pull/157
+  https://github.com/greenelab/hetmech/blob/088dc3c0852387b58880c145f265fd704eb5e5d1/explore/gamma-hurdle/parameter_estimates.ipynb
+  https://github.com/greenelab/hetmech/issues/123
+-->
+
+### Empirical DWPC p-values
```
Comment (dhimmel) [inline content/05.main-text.md:370]: @ben-heil this section here summarizes the empirical DWPC p-values implementation you helped with if you'd like to read it over.

**Comment by dhimmel** [comment]
I am going to merge and address any additional reviews subsequently.

@zietzm how is your availability? You are probably a better author than me to write the matrix DWPC & gamma-hurdle sections. Does this PR once merged provide enough of a scaffold?

**Comment by zietzm** [comment]
I've got time in the next couple days. I'll fill in some of the scaffold you've laid out

File: content/05.main-text.md (hunk 2 above)
Comment (ben-heil) [reply content/05.main-text.md:370]: Looks good to me

**Comment by cgreene** [comment]
😆  I just came in to review, but I think my latency is too low. What if you complete what you can with these rapid iterations, I will drop in if I can, but in any case I will print the whole thing out when the speed of contributions drops a bit and provide feedback at that stage.

**Comment by cgreene** [comment]
"latency is too low" ➡️ "latency is too high". Someone needs to point out that's not how latency works 😆

**Pull request merged**